src/ray/raylet/actor_registration.h

```diff
@@ -118,8 +132,10 @@ class ActorRegistration {
   /// dependencies. This is indexed by handle.
   std::unordered_map<ActorHandleID, FrontierLeaf> frontier_;
 
-  /// All of the dummy object IDs from this actor's tasks.
-  std::vector<ObjectID> dummy_objects_;
+  /// A map from all of the dummy object IDs stored for this
+  /// actor to the number of actor handles that have the object
+  /// ID in scope.
+  std::unordered_map<ObjectID, int64_t> dummy_objects_;
```

Comment:
Will we always leak one dummy object ID per actor handle? That is, the very last dummy object ID produced by the handle? If I understand correctly, the only way to release a dummy object is to extend the frontier, which happens when a new task gets submitted. But some task will be the last task.

E.g., what happens to the `dummy_objects_` map for the workload

```python
a = Actor.remote()

@ray.remote
def f(a):
    return

while True:
    f.remote(a)
```

Reply:
Yes, this is because we don't do GC for actor handles.